src/main/java/hachi/Hachi.java

```diff
@@ -0,0 +1,410 @@
+     */
+
+    public static void markOrUnmarkTask(int index, boolean isMark) {
+        taskArrayList.get(index).setCompleteness(isMark);
+        System.out.println("\tSure, I've done as you requested:");
+
+        Task currentTask = taskArrayList.get(index);
+        String taskType = currentTask.getTaskType();
+        String statusIcon = currentTask.getStatusIcon();
+        System.out.print("\t" + (index + 1) + ": ");
+        System.out.print("[" + taskType + "] ");
+        System.out.print("[" + statusIcon + "] ");
+        System.out.println(currentTask.getName());
+    }
+
+    private static void deleteTask(String cleanedInput) throws HachiException {
+        int numTasks = Task.getTotalNumTasks();
+        HachiException.checkEmptyList(numTasks);
+        int taskNumber = getDeleteTaskNumber(cleanedInput);
+        HachiException.checkOutOfBounds(taskNumber);
+
+        Task toDelete = taskArrayList.get(taskNumber - 1);
+        String taskType = toDelete.getTaskType();
+        String statusIcon = toDelete.getStatusIcon();
+        Task.decrementTotalNumTasks();
+
+        System.out.println("\tSure, I've done as you requested.");
+        System.out.print("\t[" + taskType + "] ");
+        System.out.print("[" + statusIcon + "] " + toDelete.getName());
+        System.out.println(" has been deleted from your list.");
+
+        taskArrayList.remove(taskNumber - 1);
+    }
+
+    private static int getDeleteTaskNumber(String cleanedInput) throws HachiException {
+        int indexOfTaskNum = cleanedInput.indexOf("DELETE") + 6; // find index of task number
+        int taskNumber = 0;
+
+        try {
+            taskNumber = Integer.parseInt(cleanedInput.substring(indexOfTaskNum).trim()); // parse string to int
+        } catch (NumberFormatException e){
+            HachiException.checkOutOfBounds(indexOfTaskNum);
+        }
+
+        return taskNumber;
+    }
+
+    private static int getMarkTaskNumber(String cleanedInput) throws HachiException {
+        int indexOfTaskNum = cleanedInput.indexOf("MARK") + 4; // find index of task number
+        int taskNumber = 0;
+
+        try {
+            taskNumber = Integer.parseInt(cleanedInput.substring(indexOfTaskNum).trim()); // parse string to int
+        } catch (NumberFormatException e){
+            HachiException.checkOutOfBounds(indexOfTaskNum);
+        }
+
+        return taskNumber;
+    }
+
+    /**
+     * Prints to the console a goodbye message for the user.
+     */
+
+    public static void printGoodbyeMessage() {
+        System.out.println("\tGoodbye! Hope you have a marvelous day.");
+    }
+
+    private static void loadFile(ArrayList<Task> taskArrayList) throws FileNotFoundException, HachiException {
+        File taskFile = new File("src/main/java/hachi.txt");
+        Scanner s = new Scanner(taskFile);
+
+        while (s.hasNext()) {
+            String[] line = s.nextLine().split(" \\| ");
```

Comment:
remember that plural form should be used on names representing a collection of objects :]

Reply:
Hi, can I check which is this referring to? taskFile -> tasksFile maybe?

Reply:
Hello! its referring to line 256! The 
`String[] line = s.nextLine().split(" \\| ");`

should be something like String[] lines for example or anything else more appropriate :)